core-container/entry.sh

```diff
@@ -16,88 +16,402 @@
+espw_name="espw"
+
+
+function echoerr() {
+   echo "ERROR: $1" >&2
+}
+
+
+function wait_kcp_ready() {
+    echo "Waiting for kcp to be ready... this may take a while."
+    (
+        KUBECONFIG=
+        until [ "$(kubectl logs $(kubectl get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}') -c kcp | grep '***READY***')" != "" ]; do
+            sleep 5
+        done
+    )
+    echo "Succes!"
+    echo "Copying the admin.kubeconfig from kubestellar seret..."
+    mkdir -p /home/kubestellar/.kcp
+    (
+        KUBECONFIG=
+        kubectl get secrets kubestellar -o 'go-template={{index .data "admin.kubeconfig"}}' | base64 --decode > /home/kubestellar/.kcp/admin.kubeconfig
+    )
+}
+
+
+function wait-kubestellar-ready() {
+    wait_kcp_ready
+    echo "Waiting for KubeStellar to be ready... this may take a while."
+    (
+        KUBECONFIG=
+        until [ "$(kubectl logs $(kubectl get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}') -c core | grep '***READY***')" != "" ]; do
+            sleep 5
+        done
+    )
+    echo "Succes!"
+}
+
+
+function guess_kcp_dns() {
+    KUBECONFIG=
+    if [ -z "$EXTERNAL_HOSTNAME" ]; then
+        # Try to guess the route
+        if kubectl get route kubestellar-route &> /dev/null; then
+            EXTERNAL_HOSTNAME=$(kubectl get route kubestellar-route -o yaml -o jsonpath={.spec.host} 2> /dev/null)
+        fi
     fi
-fi
-if [ -z "$EXTERNAL_HOSTNAME" ]; then
-    if kubectl get ingress kubestellar-ingress &> /dev/null; then
-        EXTERNAL_HOSTNAME=$(kubectl get ingress kubestellar-ingress -o yaml -o jsonpath={.spec.rules[0].host} 2> /dev/null)
+    if [ -z "$EXTERNAL_HOSTNAME" ]; then
+        # Try to guess the ingress
+        if kubectl get ingress kubestellar-ingress &> /dev/null; then
+            EXTERNAL_HOSTNAME=$(kubectl get ingress kubestellar-ingress -o yaml -o jsonpath={.spec.rules[0].host} 2> /dev/null)
+        fi
     fi
-fi
+    echo "${EXTERNAL_HOSTNAME}"
+}
 
-# Create the certificates
-if [ -n "$EXTERNAL_HOSTNAME" ]; then
-    echo "< Ensuring the TLS certificate >---------------------------"
-    eval pieces=($(kubestellar-ensure-kcp-server-creds ${EXTERNAL_HOSTNAME}))
-    echo "TLS certificates for server ${EXTERNAL_HOSTNAME}:"
-    echo ${pieces[0]}
-    echo ${pieces[1]}
-    echo ${pieces[2]}
-fi
 
-# Start kcp
-echo "< Starting kcp >-------------------------------------------"
+function create_or_replace() { # usage: filename
+    filename="$1"
+    kind=$(grep kind: "$filename" | head -1 | awk '{ print $2 }')
+    name=$(grep name: "$filename" | head -1 | awk '{ print $2 }')
+    if kubectl get "$kind" "$name" &> /dev/null ; then
+        kubectl replace -f "$filename"
+    else
+        kubectl create -f "$filename"
+    fi
+}
 
-mkdir -p ${PWD}/kubestellar-logs # required in Open Shift clusters
+function ensure_root_compute_configd() {
```

Comment:
Why is there a copy of material from the `kubestellar` script here? Why not invoke the script?

Reply:
@MikeSpreitzer thank you for the suggestion.
I think this is "could" ideally be achieved by running `kubestellar init` but as the script stands right now, I think  there is a previous check that I cannot pass.
Perhaps in the future we can find a way to modify the script to work for both purposes.
My quick solution was to copy part of the code.
I'm not quite sure why some of the objects are in the script and some are in the cofig folder.
Perhaps a more clean solution would be to put everything in the config.
```
# Check if kcp is running
if [ $(process_running kcp) != "running" ]
then
    echo "kcp is not running - please start it ...."
    exit 1
fi
```

Reply:
I would prefer to not merge a duplication of scripting.  Please let me know what stops this from re-using `kubestellar init` and let's get that fixed first.

Reply:
Implemented